packages/devtools_server/lib/src/handlers.dart

```diff
@@ -0,0 +1,43 @@
+import 'dart:async';
+import 'dart:isolate';
+
+import 'package:path/path.dart' as path;
+import 'package:shelf/shelf.dart' as shelf;
+import 'package:shelf/shelf_io.dart' as shelf;
+import 'package:shelf_static/shelf_static.dart';
+
+/// Default [shelf.Handler] for serving DevTools files.
+///
+/// This serves files out from the build results of running a pub build of the
+/// DevTools project.
+Future<shelf.Handler> defaultHandler() async {
+  final Uri resourceUri = await Isolate.resolvePackageUri(
+      Uri(scheme: 'package', path: 'devtools/devtools.dart'));
+  final packageDir = path.dirname(path.dirname(resourceUri.toFilePath()));
+
+  // Default static handler for all non-package requests.
+  final String buildDir = path.join(packageDir, 'build');
+  final buildHandler = createStaticHandler(
+    buildDir,
+    defaultDocument: 'index.html',
+  );
+
+  // The packages folder is renamed in the pub package so this handler serves
+  // out of the `pack` folder.
+  final String packagesDir = path.join(packageDir, 'build', 'pack');
+  final packHandler = createStaticHandler(
+    packagesDir,
+    defaultDocument: 'index.html',
+  );
+
+  // Make a handler that delegates based on path.
+  FutureOr<shelf.Response> handler(shelf.Request request) {
```

Comment:
Can you not just return this directly instead of the tear off?

Reply:
Done.